ambry-clustermap/src/main/java/com.github.ambry.clustermap/ClusterChangeHandler.java

```diff
@@ -10,42 +10,36 @@
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
  */
+
 package com.github.ambry.clustermap;
 
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import org.apache.helix.api.listeners.IdealStateChangeListener;
-import org.apache.helix.api.listeners.InstanceConfigChangeListener;
-import org.apache.helix.api.listeners.LiveInstanceChangeListener;
-import org.apache.helix.api.listeners.RoutingTableChangeListener;
-import org.apache.helix.spectator.RoutingTableSnapshot;
+import java.util.stream.Stream;
 
 
 /**
  * General handler that handles any resource or state changes in cluster. It exposes API(s) for cluster manager to
- * access up-to-date cluster info. Each data center has its own {@link ClusterChangeHandler}.
+ * access up-to-date cluster info for a data center. Each data center has its own
+ * {@link ClusterChangeHandler}.
  */
-interface ClusterChangeHandler
-    extends InstanceConfigChangeListener, LiveInstanceChangeListener, IdealStateChangeListener,
-            RoutingTableChangeListener {
+public interface ClusterChangeHandler {
   /**
    * Register a listener of cluster map for any changes.
    * @param clusterMapChangeListener the {@link ClusterMapChangeListener} to add.
    */
   void registerClusterMapListener(ClusterMapChangeListener clusterMapChangeListener);
 
   /**
-   * Set the initial snapshot in this {@link ClusterChangeHandler}.
-   * @param routingTableSnapshot the snapshot to set
-   */
-  void setRoutingTableSnapshot(RoutingTableSnapshot routingTableSnapshot);
-
-  /**
-   * @return current snapshot held by this {@link ClusterChangeHandler}.
+   * Get replicas of given partition from this datacenter that are in required state
+   * @param partition the {@link PartitionId} for which to get the list of replicas.
+   * @param state {@link ReplicaState} associated with replica
+   * @return the {@link ReplicaId}s satisfying requirements.
    */
-  RoutingTableSnapshot getRoutingTableSnapshot();
+  Stream<AmbryReplica> getReplicaIdsByState(AmbryPartition partition, ReplicaState state);
```

Comment:
Incidentally, most of the methods in ClusterChangeHandler are about getting a view of the datanodes, and replicas in that datacenter. This was the case before this change. I added this method to make the interface more general to things that do not have a helix `RoutingTableProvider`.

Perhaps a better name would be `DatacenterView` for this interface, but I would probably prefer changing the name in a separate PR since this one is already big.